pyuvsim/telescope.py

```diff
@@ -128,30 +136,47 @@ def __init__(self, beam_list=None, uvb_params=None):
         self.uvb_params.update(list_uvb_params)
         self.uvb_params.update(uvb_params)    # Optional parameters for the UVBeam objects
 
-        x_orientation_list = []
-        for beam in beam_list:
-            if self.string_mode:
-                beam_use = self._str_to_obj(beam)
-            else:
-                beam_use = beam
-
-            if hasattr(beam_use, "x_orientation"):
-                x_orientation_list.append(beam_use.x_orientation)
-        if (
-            len(x_orientation_list) == 0
-            or x_orientation_list.count(None) == len(x_orientation_list)
-        ):
-            self.x_orientation = None
-        elif all(x == x_orientation_list[0] for x in x_orientation_list):
-            self.x_orientation = x_orientation_list[0]
-        else:
-            raise ValueError("UVBeam x_orientations do not match among beams in list.")
+        self.is_consistent = False
+        if check:
+            self.check_consistency(force=force_check)
 
-        if len(x_orientation_list) > 0 and self.x_orientation is None:
-            warnings.warn(
-                "All polarized beams have x_orientation set to None. This will make it "
-                "hard to interpret the polarizations of the simulated visibilities."
-            )
+            if self.x_orientation is None:
+                warnings.warn(
+                    "All polarized beams have x_orientation set to None. This will make it "
+                    "hard to interpret the polarizations of the simulated visibilities."
+                )
+
+    @consistent
+    @property
+    def x_orientation(self):
+        """The x_orientation of all beams in list (if consistent)."""
+        return getattr(self._obj_beam_list[0], "x_orientation", None)
```

Comment:
What is returned if the beam list is not consistent? Does it error?

Reply:
Yes, it will error with `BeamConsistencyError` if not consistent. I can add that to the docstring.